src/Services/StudyAreaService.ts

```diff
@@ -982,6 +991,20 @@ module StreamStats.Services {
 
             }
         }
+
+        public computeRegressionEquation(regtype: string) {
+            this.nssService.onSelectedStatisticsGroupChanged.subscribe(new WiM.Event.EventHandler<NSSEventArgs>((sender: any, e: NSSEventArgs) => {
+                this.eventManager.SubscribeToEvent(onSelectedStudyParametersLoaded, new WiM.Event.EventHandler<StudyAreaEventArgs>((sender: any, e: StudyAreaEventArgs) => {
+                    if(e != null && e.parameterLoaded) {                    
+                        this.nssService.estimateFlows(this.studyAreaParameterList,"value", this.selectedStudyArea.RegionID, false, regtype, false);
+                        this._onQ10Loaded.raise(null, WiM.Event.EventArgs.Empty);
+                        //console.log(this.nssService.selectedStatisticsGroupList);
+                    }
+                }));
+                this.loadParameters();
+            }));
+            this.nssService.loadParametersByStatisticsGroup(this.selectedStudyArea.RegionID, "", "GC1449, GC1450", [], regtype);            
+        }

```

Comment:
Is there a reason this (and the Q10Loaded event/subscriber) is in the studyAreaService, rather than the NSSservice?

Reply:
The study area service seemed the best place because we're calculating everything for a study area and had actually set up a wateruseQ10 property on the StudyArea model initially. As we neared the end we ended up not populating that property but I'm wondering if it might be a good idea to go back and populate it anyway and then just call it in the wateruse controller.  I guess it feels like we'd be muddying the NSSService to have placed this in there, especially if we're modifying properties in the studyarea.